### ISSUE TITLE
[BC Idea]: Add event OnBeforeGetScenario in codeunit 8892 "Email Scenario Impl." in procedure GetDefaultAccount

### ISSUE THREAD

**Issue by DELNicoJdl** [microsoft/BCApps#2089]
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=e793a3bc-4a7a-ef11-a4e5-6045bdb9fcc9

### Description

Hello,

Could it be possible to add the event OnBeforeGetScenario(EmailAccount,IsHandled) in procedure GetDefaultAccount in codeunit 8892 "Email Scenario Impl."

    local procedure GetDefaultAccount(var EmailAccount: Record "Email Account")
    var
        Scenario: Record "Email Scenario";
    begin
        if not Scenario.Get(Enum::"Email Scenario"::Default) then
            exit;

        EmailAccount."Account Id" := Scenario."Account Id";
        EmailAccount.Connector := Scenario.Connector;
    end;

Become :

    local procedure GetDefaultAccount(var EmailAccount: Record "Email Account")
    var
        Scenario: Record "Email Scenario";
    begin
        OnBeforeGetScenario(EmailAccount,IsHandled);
        if IsHandled then exit;
        if not Scenario.Get(Enum::"Email Scenario"::Default) then
            exit;
    
        EmailAccount."Account Id" := Scenario."Account Id";
        EmailAccount.Connector := Scenario.Connector;
    end;

    [IntegrationEvent(false, false)]
    local procedure OnBeforeGetScenario(var EmailAccount: Record "Email Account"; var IsHandled: Boolean)
    begin
    end;

This event would able us to define a generic sender email by user.
The idea is to have one default email account per group of user.

https://github.com/microsoft/ALAppExtensions/issues/26539

### I will provide the implementation for this BC Idea

- [X] I will provide the implementation for this BC Idea

**Comment by duichwer** [comment]
Why an event and not an extension via an PR for the System Application?

**Comment by JesperSchulz** [comment]
I would agree with @duichwer here. This sounds more of a capability you should add to the email module, that it sounds like a capability you should extend the module with.
Would you be willing to contribute the capability rather than getting the extensibility point?

**Comment by DELNicoJdl** [comment]
Yes I totally agree, and I can contribute to it.

**Comment by JesperSchulz** [comment]
Approved then! Looking forward to the PR 😊

**Comment by Aleyenda** [comment]
Closed due to inactivity.
